packages/amplify_api_plugin_interface/lib/src/RestAPI/RestOptions.dart

```diff
@@ -17,22 +17,19 @@ import 'package:flutter/foundation.dart';
 import 'dart:typed_data';
 
 class RestOptions {
-  String apiName;
+  String? apiName;
   String path;
-  Uint8List body;
-  Map<String, String> queryParameters;
-  Map<String, String> headers;
+  Uint8List? body;
+  Map<String, String>? queryParameters;
+  Map<String, String>? headers;
 
   RestOptions({
     this.apiName,
-    @required this.path,
+    required this.path,
```

Comment:
So in general with these changes, we are moving issues with missing parameters from runtime to compile time? Is that correct? If so, I don't think this necessarily amounts to a 'breaking' change, but there may be some developers who find their apps suddenly don't compile, right?

Reply:
Yes that's correct.

Reply:
We should discuss this as a team.

Reply:
Confirmed with @Ashish-Nanda that we'll be doing a major version bump for this Null Safety work.